Gradle build fails with configuration cache enabled with warnings

### Expected Behavior

I expect that the configuration cache never fails the build with `warn` enabled, regardless of any issues in a plugin/configuration.

I have in `~/.gradle/gradle.properties`:

```properties
org.gradle.configuration-cache=true
org.gradle.configuration-cache.problems=warn
```

[JIB plugin](https://github.com/GoogleContainerTools/jib) defined:

```kotlin
jib {
	container {
		creationTime.set("USE_CURRENT_TIMESTAMP")
		mainClass = "com.my.MyApp"
	}
	from {
		image = "eclipse-temurin:17-jre"
	}
}
```

I run:
```shell
./gradlew jibDockerBuild
```

Build passes, I get no more than a warning from the configuration cache.

### Current Behavior

I get:
```txt
> Task :jibDockerBuild FAILED

10 problems were found storing the configuration cache, 2 of which seem unique.
- Task `:jibDockerBuild` of type `com.google.cloud.tools.jib.gradle.BuildDockerTask`: cannot deserialize object of type 'org.gradle.api.Project' as these are not supported with the configuration cache.
  See https://docs.gradle.org/8.1.1/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
- Task `:jibDockerBuild` of type `com.google.cloud.tools.jib.gradle.BuildDockerTask`: cannot serialize object of type 'org.gradle.api.internal.project.DefaultProject', a subtype of 'org.gradle.api.Project', as these are not supported with the configuration cache.
  See https://docs.gradle.org/8.1.1/userguide/configuration_cache.html#config_cache:requirements:disallowed_types

See the complete report at file:///Users/artemptushkin/gitlab/IptiQ/dx/spring-cloud-config-server/build/reports/configuration-cache/dmnth65ek47ehgnuriyy459yl/ac38wk6b6wvejhas26wg0jngi/configuration-cache-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':jibDockerBuild'.
> Cannot invoke "org.gradle.api.Project.getProjectDir()" because "this.project" is null
```

### Context (optional)

* [Gradle thread forum](https://discuss.gradle.org/t/how-to-disable-build-failures-with-enabled-configuration-cache-in-gradle/45699)
* [Jib plugin issue](https://github.com/n0mer/gradle-git-properties/issues/225)

### Steps to Reproduce

See in "expected behaviour"

### Gradle version

8.1.0

### Build scan URL (optional)

_No response_

### Your Environment (optional)

_No response_

--- a/platforms/documentation/docs/src/docs/userguide/optimizing-performance/configuration_cache.adoc
+++ b/platforms/documentation/docs/src/docs/userguide/optimizing-performance/configuration_cache.adoc
@@ -164,7 +164,7 @@ If enabled in a `gradle.properties` file, you can override that setting and disa
 === Ignoring problems
 
 By default, Gradle will fail the build if any configuration cache problems are encountered.
-When gradually improving your plugin or build logic to support the configuration cache it can be useful to temporarily turn problems into warnings.
+When gradually improving your plugin or build logic to support the configuration cache it can be useful to temporarily turn problems into warnings, with no guarantee that the build will work.
 
 This can be done from the command line:
 

